rfs/src/store/dir.rs

```diff
@@ -34,7 +34,15 @@ impl DirStore {
 #[async_trait::async_trait]
 impl Store for DirStore {
     async fn get(&self, key: &[u8]) -> Result<Vec<u8>> {
-        let path = self.root.join(hex::encode(key));
+        let file_name = hex::encode(key);
+        let dir_path = self.root.join(&file_name[0..2]);
+
+        let path = match fs::try_exists(&dir_path).await {
+            Ok(true) => dir_path.join(file_name),
+            Ok(false) => self.root.join(file_name),
```

Comment:
instead of adding this handling, why not create a once run migration script (maybe on the store new/initialization).
So we don't need to do unnecesary dir checking on each `get`

Reply:
Actually there is absolutely no need to check if dir exist or not on on a get. Trying to read the file directly is good enough. we will still get NotFound if the file or any of the directories on the path don't exist

Reply:
^ much better